### ISSUE TITLE
Receive PDFs submitted via a submission form as a Primary Document (not a Supporting File)

### ISSUE THREAD

**Issue by llfinch** [huridocs/uwazidocs#5059]
**Is your feature request related to a problem? Please describe.**
In Uwazi submission forms, we can amend the code to either accept PDFs only (`<PublicForm template="ID_OF_THE_TEMPLATE" file />`) or a wider range of media (`<PublicForm template="ID_OF_THE_TEMPLATE" attachments />`). In both of these cases, submitted things come into the entity in the Supporting Files section. This means that if a PDF is meant to be a Primary Document (so that it can take advantage of all those features like document search, references, etc), the admin user must download the PDF and reupload it as a Primary Document.

**Describe the solution you'd like**
Now that we have the ability to add a Primary Document upon creation of an entity (instead of only after it was created), it would be nice to have the option in a submission form to receive submitted PDFs directly as a Primary Document. This would eliminate the burden of having to download and reupload a PDF.

**Have you considered an alternative?**
I suppose having an in-app way to "transfer" a Supporting File to the Primary Document section with a click would work as this would also avoid the download-reupload burden. 

**Additional context**
An example from the new Uwazi demo that we're building. We have a submission form with "file" appended to the code: 

![image](https://user-images.githubusercontent.com/56080527/187667622-baa86fe3-32d2-4ced-9b6a-39a04e5c5f8b.png)

And submissions with a PDF come in like this: 

![image](https://user-images.githubusercontent.com/56080527/187667560-e7670196-8339-469b-8780-8fd4f0a1345e.png)

**Comment by RafaPolit** [comment]
I believe this has been discussed in the past and it rose questions of security, user decision at the moment of upload (how do a non-uwazi-user that is filling the form know what is a "primary document"), possible errors parsing the PDF (extracting the text, OCRing, etc.) that could render the submission "incomplete" and you end up without all the documents that the user sent.

So this is really complex.

Could it be "enough" to provide an alternative that is under the control of the curators of the submissions?  Something like a "promote to main file" button to already uploaded attachments?

I agree that the current flow of downloading - deleting -reuploading is really useless, but maybe this intermediate approach is more useful also for other scenarios?

**Comment by llfinch** [comment]
@RafaPolit for what it's worth I can see the utility in having a promote/demote button on Supporting Files and Primary Documents beyond just a submission form scenario, so I think that could be a valuable route to take. For example, thinking aloud: in the case of law libraries, say you have one entity for a law with the PDF of the law as the Primary Document. When the law is updated, it could be useful to "demote" the previous version of the law to be a Supporting File (you want to create distance to make it clear it isn't the actual version, but it's still important historical information to have) and then add a new Primary Document with the updated.

**Comment by txau** [comment]
@RafaPolit @llfinch this can be a broader conversation if you consider:
- That we want to implement #4854 to allow user to upload formats other than PDF as the main file
- We also discussed the possibility of having "viewers" for attachments

I think we should take this to the product team to maybe simplify Uwazi by removing the differentiation between attachments and main files and figure out better ways to handle versions, translations, etc.

**Comment by RafaPolit** [comment]
I am all in favor of "removing" the "main document" and simply selecting from supporting files which is the "primary viewing element" for the entity (the wording is not final, just the idea of what it should imply)

**Comment by txau** [comment]
Same as https://github.com/huridocs/uwazi/issues/5144
